lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -1932,6 +1945,21 @@ def handle_joining_data_node(self) -> None:
         if self.peers_data.get(Scope.UNIT, "started", False):
             self.status.clear(PClusterNoDataNode)
         else:
+            try:
+                config_profile = self.profiles_manager.get_config_profile()
+            except ValueError:
+                return
```

Comment:
Is this assuming that the status was already set as part of the `config_change` hook?